### DISCUSSION TITLE
JSF1091 warning error on Primefaces 13.0.1 with JSF 2.3.9

### DISCUSSION THREAD

**Discussion by clericip** [primefaces/community#2182] [PrimeFaces] [answered]
When I use Primefaces 13.0.1 with JSF 2.3.9 (with JSF 2.2 I have no warnings), I get the following warning error:
AVVERTENZA: JSF1091: No mime type could be found for file null.  To resolve this, add a mime-type mapping to the applications web.xml.

It seems that the warning occurs when I have a partial rendering response (response <?xml version='1.0' encoding='UTF-8'?><partial-response>...</partial-response>). 
The program seems to work correctly, but I get the warning JSF1091.

**Comment by melloware** [comment, marked as answer]
i see you are using Mojarra 2.3.9 have you tried 2.3.21 which is Mojarra's last 2.3.X release?  The error is coming from Mojarra not PF.

It was fixed already: https://github.com/eclipse-ee4j/mojarra/issues/4880

**Reply by clericip** [reply]
As suggested I also tried Mojarra 2.3.21 but the problem persists. Thank you.

**Reply by melloware** [reply]
Are you using Wildfly?  I have only seen this warning with Wildfly.

**Reply by clericip** [reply]
No, I use Tomcat 9. Thank you.

**Reply by melloware** [reply]
yeah weird ours went away using Wildfly but let me know if you figure it out.

**Reply by kurtomerfaruk** [reply]
Is mime-type defined in web.xml file. If you did not define mime-type have to define. this is how I solved it

**Reply by melloware** [reply]
yes try adding this to your web.xml

```xml
   <mime-mapping>
        <extension>ttf</extension>
        <mime-type>application/font-sfnt</mime-type>
    </mime-mapping>
    <mime-mapping>
        <extension>woff</extension>
        <mime-type>application/font-woff</mime-type>
    </mime-mapping>
    <mime-mapping>
        <extension>woff2</extension>
        <mime-type>application/font-woff2</mime-type>
    </mime-mapping>
    <mime-mapping>
        <extension>eot</extension>
        <mime-type>application/vnd.ms-fontobject</mime-type>
    </mime-mapping>
        <mime-mapping>
        <extension>eot?#iefix</extension>
        <mime-type>application/vnd.ms-fontobject</mime-type>
    </mime-mapping>
    <mime-mapping>
        <extension>svg</extension>
        <mime-type>image/svg+xml</mime-type>
    </mime-mapping>
    <mime-mapping>
        <extension>svg#latobold</extension>
        <mime-type>image/svg+xml</mime-type>
    </mime-mapping>
    <mime-mapping>
        <extension>svg#latoblack</extension>
        <mime-type>image/svg+xml</mime-type>
    </mime-mapping>
        <mime-mapping>
        <extension>svg#latolight</extension>
        <mime-type>image/svg+xml</mime-type>
    </mime-mapping>
    <mime-mapping>
        <extension>svg#latoregular</extension>
        <mime-type>image/svg+xml</mime-type>
    </mime-mapping>
    <mime-mapping>
        <extension>svg#fontawesomeregular</extension>
        <mime-type>image/svg+xml</mime-type>
    </mime-mapping>
```

**Reply by clericip** [reply]
It doesn't work this way either. I think the problem might be between Tomcat 9 and JSF 2.3. For now, I will stick with JSF 2.2 and in the future, I will try to deploy on TomEE. Thanks for the support.
